web/packages/teleport/src/services/resources/types.ts

```diff
@@ -167,7 +167,11 @@ export type KubernetesVerb =
   | 'portforward';
 
 export type Rule = {
-  resources?: ResourceKind[];
+  /**
+   * Typically, this is ResourceKind[], though in general, we allow
+   * unrecognized legacy resource types.
+   */
+  resources?: string[];
```

Comment:
Did we hit any issues with `resources?: ResourceKind[];`?

Reply:
Thanks for that error message. Though my question was geared towards if there were some roadblock preventing properly converting source `string[]` type to `ResourceKind[]`, sorry for not being clear. 
This looks ok to me if it works for you.

Reply:
Oh, no, the motivation was to be explicit about what can appear there. With this change, I decided that it's going to be better to be explicit about the expectations, since up until now, the non-`ResourceKind` options could appear only at runtime, and here, we start using actual non-`ResourceKind` strings in the code.